### PR TITLE
User microposts

### PR DESCRIPTION
**Pull request by agalkiewicz**

### CODE REVIEW

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  private
+
+    def micropost_params
+      params.require(:micropost).permit(:content, :picture)
+    end
+
+    def correct_user
+      @micropost = current_user.microposts.find_by(id: params[:id])
```
Comment (mmarcinkowski) [inline app/controllers/microposts_controller.rb]: `.find_by_id(params[:id])`

File: app/models/user.rb [hunk 2]
```diff
@@ -56,6 +57,10 @@ def password_reset_expired?
     reset_sent_at < 2.hours.ago
   end
 
+  def feed
+    Micropost.where("user_id = ?", id)
```
Comment (mmarcinkowski) [inline app/models/user.rb]: `Micropost.where(user_id: id)` or even add a new scope to Micropost model

File: Gemfile [hunk 3]
```diff
@@ -3,6 +3,9 @@ source 'https://rubygems.org'
 gem 'rails',                   '4.2.2'
 gem 'bcrypt',                  '3.1.7'
 gem 'faker',                   '1.4.2'
+gem 'carrierwave',             '0.10.0'
+gem 'mini_magick',             '3.8.0'
+gem 'fog',                     '1.36.0'
```
Comment (mmarcinkowski) [inline Gemfile]: do u need this gem ?

File: app/uploaders/picture_uploader.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class PictureUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+  process resize_to_limit: [400, 400]
+
+  storage :file
```
Comment (mmarcinkowski) [inline app/uploaders/picture_uploader.rb]: u dont use `:fog` option here, so i guess `fog` gem can be removed

File: app/views/shared/_micropost_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  <div class="field">
+    <%= f.text_area :content, placeholder: "Compose new micropost..." %>
+  </div>
+  <%= f.submit "Post", class: "btn btn-primary" %>
+  <span class="picture">
+    <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png" %>
+  </span>
+<% end %>
+
+<script type="text/javascript">
```
Comment (mmarcinkowski) [inline app/views/shared/_micropost_form.html.erb]: move it to separate js file

File: app/views/static_pages/home.html.erb [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
+    </h2>
 
-  <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
-</div>
+    <%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
+  </div>
 
-<%= link_to image_tag("rails.png", alt: "Rails logo"),
-            'http://rubyonrails.org/' %>
+  <%= link_to image_tag("rails.png", alt: "Rails logo"), 'http://rubyonrails.org/' %>
```
Comment (mmarcinkowski) [inline app/views/static_pages/home.html.erb]: quotation marks :wink:

File: app/views/shared/_micropost_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  $('#micropost_picture').bind('change', function() {
```
Comment (mmarcinkowski) [inline app/views/shared/_micropost_form.html.erb]: add `js-` prefix to the name`

File: Gemfile (hunk 3 above)
Comment (agalkiewicz) [inline Gemfile]: I forgot to delete it. Couldn't sign up on Amazon Web Services and didn't do cloud storage in production.

File: app/uploaders/picture_uploader.rb (hunk 4 above)
Comment (agalkiewicz) [inline app/uploaders/picture_uploader.rb]: Right.

File: app/views/static_pages/home.html.erb (hunk 6 above)
Comment (agalkiewicz) [inline app/views/static_pages/home.html.erb]: Done.

File: app/models/user.rb (hunk 2 above)
Comment (agalkiewicz) [inline app/models/user.rb]: Done.

File: app/controllers/microposts_controller.rb (hunk 1 above)
Comment (agalkiewicz) [inline app/controllers/microposts_controller.rb]: Done.

File: app/views/shared/_micropost_form.html.erb (hunk 5 above)
Comment (agalkiewicz) [inline app/views/shared/_micropost_form.html.erb]: Done.

File: app/views/shared/_micropost_form.html.erb (hunk 7 above)
Comment (agalkiewicz) [inline app/views/shared/_micropost_form.html.erb]: Done.

File: app/assets/javascripts/micropost_picture_size.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+$('#js-micropost_picture').bind('change', function() {
```
Comment (mmarcinkowski) [inline app/assets/javascripts/micropost_picture_size.js]:
```
$(document).ready(function() {
  $('#js-...').bind({});
});
```

**Comment by mmarcinkowski** [comment]
except this one comment it looks ok ;)

**Pull request merged**